src/libbson/src/bson/bson-utf8.c

```diff
@@ -278,6 +280,14 @@ bson_utf8_escape_for_json (const char *utf8, /* IN */
    end = utf8 + utf8_len;
 
    while (utf8 < end) {
+      // Check if expected char length goes past end
+      _bson_utf8_get_sequence (utf8, &length_of_char, &mask);
+      if (utf8 + length_of_char > end) {
```

Comment:
```suggestion
      if (utf8 > end - length_of_char) {
```
Avoid forming invalid out-of-range pointers (pointer equivalent to integer overflow).